js/views/modals/about/Donations.js

```diff
@@ -1,69 +1,107 @@
-import app from '../../../app';
+import { clipboard } from 'electron';
+import qr from 'qr-encode';
+import { isSupportedWalletCur } from '../../../data/walletCurrencies';
 import { getWallet, launchWallet } from '../../../utils/modalManager';
-import { openSimpleMessage } from '../../modals/SimpleMessage';
 import loadTemplate from '../../../utils/loadTemplate';
 import baseVw from '../../baseVw';
-import { clipboard } from 'electron';
-import qr from 'qr-encode';
 
-const obDonationAddress = '3EN8kP3yW9MGvyiRMtGqKavQ9wYg4BspzR';
-const qrCodeDataURI = qr(`bitcoin:${obDonationAddress}`, { type: 6, size: 6, level: 'Q' });
-const donationCountFloor = 500;
+
 let hiderTimer;
 
 export default class extends baseVw {
   constructor(options = {}) {
+    const opts = {
+      ...options,
+      initialState: {
+        showCoin: 'BTC',
+      },
+      ...options.initialState,
+    };
+
     super({
       className: 'aboutDonations',
-      ...options,
+      ...opts,
     });
+    this.options = opts;
+
+    const btcAddress = '3EN8kP3yW9MGvyiRMtGqKavQ9wYg4BspzR';
+    const bchAddress = 'prws8awqjg8l497x3h7qeu6jps4fdmk75vhlng8akj';
+
+    this.dCoins = {
+      BTC: {
+        obDonationAddress: btcAddress,
+        qrCodeDataURI: qr(`bitcoin:${btcAddress}`,
+          { type: 6, size: 6, level: 'Q' }),
+        supported: isSupportedWalletCur('BTC'),
+      },
+      BCH: {
+        obDonationAddress: bchAddress,
+        qrCodeDataURI: qr(`bitcoin:${bchAddress}`,
+          { type: 6, size: 6, level: 'Q' }),
+        supported: isSupportedWalletCur('BCH'),
```

Comment:
Can this be `walletSupported` or something less generic?

Reply:
Also, I think you'd want to conditionally pass in the testnet code if we're on testnet, otherwise the buttons will always be false on testnet.

You could hard-code the mainnet code with a `T` preface if you want. But, that breaks the whole motif where the testnet code is driven by thee config. So, to keep that in play, you could make a `ensureTestnetCode` as an inverse of `ensureMainnetCode`. But, completely up to you - it may be overkill - as I'm pretty sure the server just mashes a `T` onto a mainnet code to get testnet.

Reply:
We want it to always be false on testnet. You can't send donations from a testnet wallet.